### PR TITLE
better default gulp targets

### PR DESCRIPTION
**Pull request by nickboldt**
add package-simple, package-bundle, and dist; update README and
package.json; remove create-zip (not needed); by default do 'run', not build.

### CODE REVIEW

**Comment by nickboldt** [comment]
with the above changes, you can produce the stub installer and the bundle installer like this:

    gulp
    mv dist/win/*.exe dist/win/Setup-stub.exe
   gulp package-bundle
   mv dist/win/*.exe dist/win/Setup-bundle.exe

But if someone merges https://github.com/redhat-developer-tooling/developer-platform-install/pull/74 then the two move steps are no longer needed.

**Comment by ppitonak** [comment]
This PR looks good, the only thing I don't like is "stub" in name of new Gulp task. @nickboldt @dgolovin @jrichter1 any suggestions for better name?

**Comment by nickboldt** [comment]
How about package-simple and package-bundle? I could also add a package-all that would build both in a single default operation (clean, generate + package both installers, then cleanup 7zip leftovers).

Note that doing both in one task requires that someone push https://github.com/redhat-developer-tooling/developer-platform-install/pull/74 so that the output files from each package task don't overwrite each other.

**Comment by ppitonak** [comment]
I'm testing #74 right now, download of all deps takes too long :(

**Comment by nickboldt** [comment]
After this change, running just 'gulp' gets this:

$➔ ll dist/win/
total 1.6G
-rwxr-xr-x 1 nboldt 197609  37M Apr  6 11:24 DeveloperPlatformInstaller-win32-x64.exe
drwxr-xr-x 1 nboldt 197609    0 Apr  6 11:24 DeveloperPlatformInstaller-win32-x64
-rwxr-xr-x 1 nboldt 197609 1.6G Apr  6 11:50 DeveloperPlatformInstaller-win32-x64-bundle.exe

**Comment by dgolovin** [comment]
I got both installers 1.6G after building.

**Comment by ppitonak** [comment]
1. I suggest that we rename _packages_ task to _dist_
2. you should update npm scripts in package.json to contain start (default task), package-simple, package-bundle and dist
3. you should update Readme
3. I suggest that we change default task to start _run_ task as this is what people probably run during development

**Comment by nickboldt** [comment]
Should the default task be "start" or "run" ? You appear to be asking for both.

File: gulpfile.js [hunk 1]
```diff
@@ -126,9 +119,19 @@ gulp.task('package', function (cb) {
   });
 });
 
-// Create bundled installer
+// Create stub installer that will then download all the requirements
```
Comment (dgolovin) [inline gulpfile.js]: all package* tasks should have clean as first task in sequence.

File: gulpfile.js [hunk 2]
```diff
@@ -170,3 +169,7 @@ gulp.task('prefetch', function(cb) {
   }
   artifactType = "-bundle";
 });
+
+gulp.task('default', function() {
```
Comment (dgolovin) [inline gulpfile.js]: Should be

``` JavaScript
gulp.task('default', ['run'], function() {});
```

File: gulpfile.js (hunk 1 above)
Comment (nickboldt) [inline gulpfile.js]: I disaagree. Why do you need to clean again after creating the first installer?

File: gulpfile.js (hunk 2 above)
Comment (nickboldt) [inline gulpfile.js]: But transpile (which is what the run task does) is meaningless w/o first doing a generate, right?

File: gulpfile.js (hunk 1 above)
Comment (ppitonak) [inline gulpfile.js]: I think that you are not talking about the same thing :) 

1. clean should definitely run before task _package-simple_ and _package-bundle_
2. in task _packages_, it should run as first task, doesn't need to run after first installer is created

File: gulpfile.js (hunk 2 above)
Comment (ppitonak) [inline gulpfile.js]: This is what I asked for in my comment

    gulp.task('default', ['run']);  // nothing more is necessary

Transpile creates main/index.js, it has nothing to do with generate

**Comment by nickboldt** [comment]
Refactored again. Is this what you both want? Wasn't sure about adding "start" to package.json and if we still need to include a "package":"gulp package" line.

**Comment by ppitonak** [comment]
Package.json already contains _start_ script and it calls `gulp run`, I'm fine with having it this way. For the future, please verify that Markdown files render correctly ;)

Merged. Thanks for your PR.

**Pull request closed without merging**